### ISSUE TITLE
Kitty rendering nvim text incorrectly, fixed with `kitty --config NONE`

### ISSUE THREAD

**Issue by Dhamma-valence** [kovidgoyal/kitty#4370]
**Describe the bug**
When using neovim inside kitty the screen isn't redrawn properly and leaves artifacts on the screen when scrolling, when something is pasted, basically when anything changes except for just typing. It works as expected in Alacritty but seems to be broken in Kitty.

**To Reproduce**
1. Get a file with text in it, either by pasting something into it, or opening a file that already has text in it and scrolling.
2. The text is seemingly randomly broken up and unusable.

**Screenshots**
https://imgur.com/a/1jgpb7g

**Environment details**
```
kitty 0.23.1 (0456399ce5) created by Kovid Goyal
Linux $HOSTNAME 5.15.8-xanmod1-tt #0~git20211214.32581e4 SMP Tue Dec 14 17:19:02 UTC 2021 x86_64
Debian GNU/Linux bookworm/sid $HOSTNAME pts17

Running under:X11
Frozen: False
Paths:
  kitty: /home/$USER/src/kitty/kitty/launcher/kitty
  base dir: /home/$USER/src/kitty
  extensions dir: /home/$USER/src/kitty/kitty
  system shell: /bin/zsh-static
Loaded config files:
  /home/$USER/.config/kitty/kitty.conf

Config options different from defaults:
clipboard_control ('write-clipboard', 'write-primary')
Changed mouse actions:
	 b1 -2 ungrabbed mouse_click_url_or_select
	 shift+b1 -2 ungrabbed mouse_click_url_or_select
	 shift+b1 -2 grabbed mouse_click_url_or_select
	 shift+ctrl+b1 repeat ungrabbed mouse_click_url
	 shift+ctrl+b1 repeat grabbed mouse_click_url
Added shortcuts:
	shift+super+w close_os_window

```

**Additional context**
The problem does seem to go away when running kitty with `kitty --config NONE`. [Here is the current config](https://gist.github.com/Dhamma-valence/c7c1d3377c62a094862546b7b7f6fd0b)

**Comment by kovidgoyal** [comment]
I can see nothing broken in your screenshots. The only difference between kitty and alacritty seems to be in the latter there are hard line breaks inserted after every line. And I certainly cannot reproduce anything broken by scrolling around in nvim in kitty. You are going to need to provide more details than this.

**Comment by Dhamma-valence** [comment]
Hopefully these videos I made will make it a little clearer:
https://imgur.com/a/eeeHSDo

I figured out what must've caused it, it works fine with the default config but after running `kitty +kitten themes` and setting it to any of the themes causes this bug. Just the default, or NONE configs works fine. I'm going to try manually editing the config file to change the theme to one of the different ones instead of using the `kitty +kitten themes` tool, though I'm not sure how that's different from just using the tool.
